python/cudf/cudf/core/series.py

```
@@ -815,6 +817,100 @@ def __array_function__(self, func, types, args, kwargs):
+        dtype: object
+
+        ``map`` accepts a ``dict`` or a ``Series``. Values that are not found
+        in the ``dict`` are converted to ``NaN``, unless the dict has a default
+        value (e.g. ``defaultdict``):
+
+        >>> s.map({'cat': 'kitten', 'dog': 'puppy'})
+        0   kitten
+        1    puppy
+        2      NaN
+        3      NaN
+        dtype: object
+
+        It also accepts numeric functions:
+        >>> s = cudf.Series([1, 2, 3, 4, np.nan])
+        >>> s.map(lambda x: x ** 2)
+        0       1
+        1       4
+        2       9
+        3       16
+        4       NaN
+        dtype: int64
+
+        Please note map does not currently support user defined functions
+        with dtypes `str` and`category`.
+        """
+        if isinstance(arg, dict):
+            lhs = cudf.DataFrame({"x": self, "orig_order": arange(len(self))})
+            rhs = cudf.DataFrame(
+                {
+                    "x": arg.keys(),
+                    "s": arg.values(),
+                    "bool": full(len(arg), True, dtype=self.dtype),
+                }
+            )
+            res = lhs.merge(rhs, on="x", how="left").sort_values(
+                by="orig_order"
+            )
+            result = res["s"]
+            result.name = self.name
+            result.index = self.index
+        elif isinstance(arg, cudf.Series):
+            lhs = cudf.DataFrame({"x": self, "orig_order": arange(len(self))})
+            rhs = cudf.DataFrame(
+                {
+                    "x": arg.keys(),
+                    "s": arg,
+                    "bool": full(len(arg), True, dtype=self.dtype),
+                }
+            )
+            res = lhs.merge(rhs, on="x", how="left").sort_values(
+                by="orig_order"
+            )
+            result = res["s"]
+            result.name = self.name
+            result.index = self.index
+        elif isinstance(self._column, cudf.core.column.CategoricalColumn):
```

I think we can just allow `applymap` to fail against unsupported types. If it's not failing cleanly / gracefully then that's something we should raise an issue for and tackle in a follow up PR.

---

Sounds good!